src/users/users.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@nestjs/common'
+import { Pageable, QueryUserDto } from './dto/get-user.dto'
+import { UserEntity } from './entities/user.entity'
+import { UsersRepository } from './users.repository'
+
+@Injectable()
+export class UsersService {
+    constructor(private usersRepository: UsersRepository) {}
+
+    async getUserById(id: number, returnStudent: boolean): Promise<UserEntity> {
+        return await this.usersRepository.getUserById(id, returnStudent)
+    }
+
+    async getManyUsers(queryUserDto: QueryUserDto): Promise<Pageable<UserEntity>> {
+        const { page, pageSize, ...filter } = queryUserDto
+
+        // map to like query
+        const likeFilter: Record<string, string> = Object.keys(filter).reduce((acc, key) => {
+            if (filter[key as keyof typeof filter]) {
+                return {
+                    ...acc,
+                    [key]: {
+                        contains: filter[key as keyof typeof filter],
```

Comment:
every value search by contains ?

Reply:
Will fix it 
this is value that I think it need to use contains

```
username
firstnameTh
lastnameTh
firstnameEn
lastnameEn
email
```

 Do you agree?

Reply:
Ok kub